src/UIKit/UIGestureRecognizer.cs

```diff
@@ -30,18 +34,26 @@ public UIGestureRecognizer (Action action) : this (Selector.GetHandle (tsel), ne
 		{
 		}
 
+		// Called by the Dispose() method
+		void OnDispose ()
+		{
+			foreach (var kv in recognizers)
+				RemoveTarget (kv.Key, kv.Value);
+			recognizers = null;
+		}
+		
 		//
 		// Signature swapped, this is only used so we can store the "token" in recognizers
 		//
 		public UIGestureRecognizer (Selector sel, Token token) : this (token, sel)
 		{
-			recognizers = token;
+			recognizers [token] = sel.Handle;
```

Comment:
that can cause an NRE here since the check (in `this`) would be done later.

Reply:
You mean because the selector might be null?   I do not follow the "would be done later" part

Reply:
This would call the constructor generated for the init: method, and that one should throw already if either sel or token are null in this case.

Reply:
yes, the input can be `null`, directly or indirectly, e.g. `Selector.FromHandle` can return `null`

`new UIGestureRecognizer (null, token);` would throw the `NullReferenceException` before it calls `this (token, sel)` which has the null check (and would throw an `ArgumentNullException`)

Reply:
@spouliot the other (`this`) constructor is called first, not last.